ambari-server/src/main/java/org/apache/ambari/server/topology/AmbariContext.java

```diff
@@ -710,26 +712,21 @@ private void createConfigGroupsAndRegisterHost(ClusterTopology topology, String
     // iterate over topo host group configs which were defined in
     for (Map.Entry<String, Map<String, String>> entry : userProvidedGroupProperties.entrySet()) {
       String type = entry.getKey();
-      String service = stack.getServicesForConfigType(type)
-        .filter(each -> topology.getServiceTypes().contains(each))
-        .findFirst()
-        // TODO check if this is required at all (might be handled by the "orphan" removal)
-        // TODO move this validation earlier
-        .orElseThrow(() -> new IllegalArgumentException("Specified configuration type is not associated with any service in the blueprint: " + type));
-
-      Config config = configFactory.createReadOnly(type, groupName, entry.getValue(), null);
-      //todo: attributes
-      Map<String, Config> serviceConfigs = groupConfigs.get(service);
-      if (serviceConfigs == null) {
-        serviceConfigs = new HashMap<>();
-        groupConfigs.put(service, serviceConfigs);
-      }
-      serviceConfigs.put(type, config);
+      List<Pair<StackId, String>> stackServices = stack.getStackServicesForConfigType(type).
+        filter(each -> topology.getServiceTypes().contains(each.getValue())).collect(Collectors.toList());
+      Preconditions.checkArgument(!stackServices.isEmpty(), new IllegalArgumentException("Specified configuration type is not associated with any service in the blueprint: " + type));
```

Comment:
Passing the message to `checkArgument` would be enough, no need to create the exception.  Something like:

```
checkArgument(!stackServices.isEmpty(),
  "Specified configuration type is not associated with any service in the blueprint: %s",
  type));
```

Reply:
Thanks.